### PR TITLE
Added New features - user access mgmt, service connectors to confluent .net sdk

### PR DESCRIPTION
**Pull request by DeepikaNMS**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -1309,6 +1309,13 @@
         "NFVIs",
         "nfvis"
       ]
+    },
+    {
+      "filename": "**/sdk/confluent/**/*.cs",
+      "words": [
+        "Acces",
```
Comment (jsquire) [inline .vscode/cspell.json:1316]: This is not an actual word; please fix the code rather than adding misspellings to the global file.

**Review by jsquire — DISMISSED** [review]

File: sdk/confluent/Azure.ResourceManager.Confluent/api/Azure.ResourceManager.Confluent.netstandard2.0.cs [hunk 2]
```diff
@@ -14,6 +14,8 @@ public static partial class ConfluentExtensions
...
         public static System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Confluent.ConfluentOrganizationResource>> ValidateOrganizationAsync(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource, string organizationName, Azure.ResourceManager.Confluent.ConfluentOrganizationData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.Response<Azure.ResourceManager.Confluent.Models.ValidationResponse> ValidateOrganizationV2Validation(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource, string organizationName, Azure.ResourceManager.Confluent.ConfluentOrganizationData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (jsquire) [inline sdk/confluent/Azure.ResourceManager.Confluent/api/Azure.ResourceManager.Confluent.netstandard2.0.cs:17]: @m-nash, @ArthurMa1978:  This doesn't seem like a public name that would pass review.

**Comment by DeepikaNMS** [comment]
Created issue for spell check issue : https://github.com/Azure/autorest.csharp/issues/4291#issue-2158464258
@jsquire

File: sdk/confluent/Azure.ResourceManager.Confluent/api/Azure.ResourceManager.Confluent.netstandard2.0.cs (hunk 2 above)
Comment (DeepikaNMS) [reply sdk/confluent/Azure.ResourceManager.Confluent/api/Azure.ResourceManager.Confluent.netstandard2.0.cs:17]: @m-nash , @ArthurMa1978  : can you pl. check on this

**Review by ArthurMa1978 — CHANGES_REQUESTED** [review]
This stable release requires further refinement.

File: sdk/confluent/Azure.ResourceManager.Confluent/api/Azure.ResourceManager.Confluent.netstandard2.0.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/confluent/Azure.ResourceManager.Confluent/api/Azure.ResourceManager.Confluent.netstandard2.0.cs:17]: Yes, it's fixed

File: sdk/confluent/Azure.ResourceManager.Confluent/tests/ConfluentManagementTestBase.cs [hunk 3]
```diff
@@ -23,19 +29,29 @@ protected ConfluentManagementTestBase(bool isAsync)
             : base(isAsync)
         {
         }
-
         [SetUp]
-        public void CreateCommonClient()
+        public async Task CreateCommonClient()
         {
             Client = GetArmClient();
+            DefaultSubscription = await Client.GetSubscriptions().GetAsync("ff9490e3-e714-4d26-b33b-4deefb0d5ffa");
```
Comment (ArthurMa1978) [inline sdk/confluent/Azure.ResourceManager.Confluent/tests/ConfluentManagementTestBase.cs:36]: Don't use fixed subs

File: sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+            Assert.NotNull(confluentNewOrg.Value.Id);
+            Assert.NotNull(confluentNewOrg.Value.Data.SsoUri);
+        }
+
+        [RecordedTest]
+        [TestCase]
+        public async Task ListConfluentOrgsInSubscription()
+        {
+            int count = 0;
+            await foreach (ConfluentOrganizationResource confluentOrganization in DefaultSubscription.GetConfluentOrganizationsAsync())
```
Comment (ArthurMa1978) [inline sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs:48]: All test cases should be re-runnable, so here need create a resource first, then iterate all resources and verify if the resource was created.

File: sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                ++count;
+            }
+            Assert.GreaterOrEqual(count, 1);
+        }
+
+        [RecordedTest]
+        [TestCase]
+        public async Task GetEnvironmentsInOrg()
+        {
+            await foreach (ConfluentOrganizationResource confluentOrganization in DefaultSubscription.GetConfluentOrganizationsAsync())
```
Comment (ArthurMa1978) [inline sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs:60]: All test cases should be re-runnable, no assumption of any pre-created resources existing.

File: sdk/confluent/Azure.ResourceManager.Confluent/tests/ConfluentManagementTestBase.cs (hunk 3 above)
Comment (DeepikaNMS) [reply sdk/confluent/Azure.ResourceManager.Confluent/tests/ConfluentManagementTestBase.cs]: i have updated

File: sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs (hunk 4 above)
Comment (DeepikaNMS) [reply sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs]: Yes followed this one.

File: sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs (hunk 5 above)
Comment (DeepikaNMS) [reply sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs]: Updated

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**